### ISSUE TITLE
Alternate to Issue #9

### ISSUE THREAD

**Issue by alexagosto** [alexagosto/Ciic4030-Project#11]
Issue #9 is based around the PLY lexer and parser, which is currently in use on both main branch and the creating-functional-code branch.

The new branch "Personal-lexer-and-parser" and subsequent (yet to be made) "personal-interpreter" branch that will be forked from PLaP branch have a custom lexer parser and interpreter to achieve the same result. 

This is an alternative and both options should be considered. Currently working on this new issue until i run into too many problems and go back to issue #9 

DO NOT MERGE EITHER BRANCH WITH MAIN UNTIL DECISION IS MADE ABOUT WHICH SOLUTION TO GO WITH.

**Comment by alexagosto** [comment]
Now has included interpreter merged into first branch, renamed to custom lexer parser and interpreter.
Has support for exponents, next commits will issue fixes to adding variables
